aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsFluentClientDecorator.kt

```diff
@@ -138,15 +138,14 @@ private class AwsFluentClientExtensions(types: Types) {
                     C: #{SmithyConnector}<Error = E> + Send + 'static,
                     E: Into<#{ConnectorError}>,
                 {
-                    let retry_config = conf.retry_config.as_ref().cloned().unwrap_or_default();
-                    let timeout_config = conf.timeout_config.as_ref().cloned().unwrap_or_default();
-                    let sleep_impl = conf.sleep_impl.clone();
+                    let retry_config = conf.retry_config().cloned().unwrap_or_default();
```

Comment:
Not blocking since it was already doing this before: it seems like `timeout_config` and `retry_config` could both be `pub(crate)` and then we could avoid the clones since we already have ownership of the `Config` struct.

Reply:
We can't do that b/c we need to use the `Config` struct to build the service client. Only part of it is used to create the inner smithy client.